reaper/reaper.go

```diff
@@ -371,8 +488,51 @@ func allReapables() (map[string][]reaperevents.Reapable, []reaperevents.Reapable
 			}
 		}
 	}
-	if config.Snapshots.Enabled {
 
+	// get all instances
+	for i := range getInstances() {
+		// add security groups to map of in use
+		for id, name := range i.SecurityGroups {
```

Comment:
set bools for instance if it is autoscaled, in a CF, if it has SGs mark them as dependencies